File: 30-Days-of-Code/Day 20 - Sorting/Solution.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    
    int n;
    scanf("%d", &n);
    
    int numSwaps=0;
    int arr[n];     
    
    for (int i = 0; i < n; i++){
        scanf("%d", &arr[i]);
    }
    
    for (int i = 0; i < n; i++){
        for (int j = 0; j < n - 1; j++){
            if(arr[j] > arr[j + 1]){
                int tmp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = tmp;
                numSwaps++;
            }
        }
        
        if (numSwaps == 0) break;
        
    }
    
    printf("Array is sorted in %d swaps.\n", numSwaps);
    printf("First Element: %d\n", arr[0]);
    printf("Last Element: %d\n", arr[n-1]);
    
    return 0;
    
}

File: 30-Days-of-Code/Day 20 - Sorting/Solution.py
import java.util.Scanner;

public class Solution {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        
        int n = in.nextInt();
        int arr[] = new int[n];
        
        for (int i = 0; i < n; i++) {
            arr[i] = in.nextInt();
        }

        int num_swaps = 0;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    int tmp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = tmp;
                    num_swaps++;
                }
            }

            if (num_swaps == 0) {
                break;
            }
        }

        System.out.println("Array is sorted in " + num_swaps + " swaps.");
        System.out.println("First Element: " + arr[0]);
        System.out.println("Last Element: " + arr[arr.length - 1]);
        
    }
}